jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -70,23 +76,43 @@ def _parse_image(app, file_obj, data_label, show_in_viewer, ext=None):
     if isinstance(file_obj, fits.HDUList):
         if 'ASDF' in file_obj:  # JWST ASDF-in-FITS
             if HAS_JWST_ASDF:
-                data_iter = _jwst_to_glue_data(file_obj, ext, data_label)
+
+                # Load all extensions
+                if ext is not None and ext in ['*']:
+                    data_iter = _jwst_all_to_glue_data(file_obj, data_label)
+
+                # Load only specified extension
+                else:
+                    data_iter = _jwst_to_glue_data(file_obj, ext, data_label)
+
+                    # if more than one viewable extension is found in the file,
+                    # issue info message.
+                    _info_nextensions(app, file_obj)
+
             else:
                 raise ImportError('jwst package is missing')
 
-        elif ext is not None:  # Load just the EXT user wants
+        elif ext is not None and ext in ['*']:  # Load all extensions
+            data_iter = _hdus_to_glue_data(file_obj, data_label)
+
+        elif ext is not None and ext not in ['*']:  # Load just the EXT user wants
             hdu = file_obj[ext]
             _validate_fits_image2d(hdu)
             data_iter = _hdu_to_glue_data(hdu, data_label, hdulist=file_obj)
 
         else:  # Load first image extension found
-            found = False
+            found = 0
             for hdu in file_obj:
                 if _validate_fits_image2d(hdu, raise_error=False):
                     data_iter = _hdu_to_glue_data(hdu, data_label, hdulist=file_obj)
-                    found = True
+
+                    # if more than one viewable extension is found in the file,
+                    # issue info message.
+                    found = _info_nextensions(app, file_obj)
```

Comment:
Actually, the way you implemented this, we don't need `found` to keep count here after all. Probably better to turn `found` back into `bool`. `_info_nextensions` does not need to return anything.

Reply:
`found` is boolean now.